examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -224,31 +226,35 @@ CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpa
     }
     else
     {
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(1) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        /* Get nocsr */
-        ByteSpan csr_data;
-        msg_reader.Get(csr_data);
-        uint8_t tmp = csr_data.size();
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+        uint8_t csrlen = tagReader.GetLength();
         /* Set nocsr length */
-        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
-        /* Set nocsr data */
-        se05x_delete_key(NOCSR_CSR_DATA_SE05X_ID);
-        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_DATA_SE05X_ID, csr_data.data(), csr_data.size()),
-                            CHIP_ERROR_INTERNAL);
-
-        msg_reader.Next();
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(2) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        /* Get nocsr nonce */
-        ByteSpan nocsr_nonce;
-        msg_reader.Get(nocsr_nonce);
-        /* Set nocsr nonce data */
-        VerifyOrReturnError(CHIP_NO_ERROR ==
-                                se05xSetCertificate(NOCSR_CSR_NONCE_DATA_SE05X_ID, nocsr_nonce.data(), nocsr_nonce.size()),
-                            CHIP_ERROR_INTERNAL);
-
-        if (message_to_sign.size() >= 16)
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_LEN_SE05X_ID, &csrlen, 1), CHIP_ERROR_INTERNAL);
+        if (csrlen > 0)
+        {
+            /* Get nocsr */
+            ByteSpan csr_data;
+            ReturnErrorOnFailure(tagReader.Get(csr_data));
+            /* Set nocsr data */
+            se05x_delete_key(NOCSR_CSR_DATA_SE05X_ID);
+            VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_DATA_SE05X_ID, csr_data.data(), csr_data.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t noncelen = tagReader.GetLength();
+        if (noncelen > 0)
+        {
+            /* Get nocsr nonce */
+            ByteSpan nocsr_nonce;
+            ReturnErrorOnFailure(tagReader.Get(nocsr_nonce));
+            /* Set nocsr nonce data */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(NOCSR_CSR_NONCE_DATA_SE05X_ID, nocsr_nonce.data(), nocsr_nonce.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        if (message_to_sign.size() - (csrlen + noncelen + 6 /* Tag + control byte + len */ + 2 /* start and end containers*/) >= 16)
```

Comment:
Again, better to use the reader's actual position than this guess?

Reply:
I see some issues when i use the Next() for parsing the TLV data here. Using Find() utility to get the TLV data. For this I am recomputing the value data utilized.